src/mjoindices/empirical_orthogonal_functions.py

```diff
@@ -608,22 +608,28 @@ def load_original_eofs_for_doy(dirname: Path, doy: int) -> EOFData:
     return EOFData(orig_lat, orig_long, eof1, eof2)
 
 
-def load_all_eofs_from_directory(dirname: Path, no_leap_years: bool) -> EOFDataForAllDOYs:
+def load_all_eofs_from_directory(dirname: Path) -> EOFDataForAllDOYs:
     """
     Loads the EOF functions (created with the function :func:`EOFDataForAllDOYs.save_all_eofs_to_dir`)
     for all DOYs from the given directory
 
     :param dirname: The directory in which the files are stored.
-    :param no_leap_years: True if every year has 365 days, False if dataset contains leap years. 
 
     :return: The EOFs for all DOYs.
     """
     eofs = []
-    for doy in doy_list(no_leap_years):
+    for doy in doy_list(no_leap_years=False):
         filename = dirname / Path("eof%s.txt" % format(doy, '03'))
+        if doy == 366:
```

Comment:
Great that you already did this automatic detection. 

I think it will work fine (as you also tested, I think), but maybe the code is somewhat hard to read and could be slightly improved

Generally, I would prefer to have the return statement as the last line of the function.

Without thinking too long, I guess there are two ways to achieve this
1) You do the try: except: for each doy and distinguisch doy 366 only in the except block. If any other doy crashes, it is an error in the data for sure, if 366 crashes, you warn the user that the software takes this as no_leap_year config

2) you put the final assignment into an else block
       else: 
            eof = load_single_eofs_from_txt_file(filename)
            eofs.append(eof)